### DISCUSSION TITLE
Charon CMD

### DISCUSSION THREAD

**Discussion by jhay06** [strongswan/strongswan#1374] [Q&A]
Can charon cmd be build on android system as a .so library or a binary ? 

I would like to  include charon cmd only not a whole strongswan   on my project ? 

Is this possible  ? 

Thanks

**Comment by jhay06** [comment]
im sorry this is not a bug

**Comment by tobiasbrunner** [comment]
To what end? `charon-cmd` won't be able to operate as a VPN client without root privileges. Have you looked at our [Android app](https://docs.strongswan.org/docs/5.9/os/androidVpnClient.html)?

**Reply by Thermi** [reply]
Maybe for loading it as a library and using all the code parts it has together with libstrongswan?
